### ISSUE TITLE
Validate that Dagster Enterprise plan is in fact better for sensors

### ISSUE THREAD

**Issue by evantahler** [airbytehq/airbyte#34368]
More deets in slack https://airbytehq-team.slack.com/archives/C059CTCUPDL/p1705615802664099

Our hypothesis is that the sensor startup was taking all the runtime, not the sensor itself (due to for example the resources that we were loading up)? The outcome being that the ways to solve this are
1. Reduce (coded) resources required for the sensor code (how we mitigated)
2. Add (infra) resources for the sensor itself so loading this stuff doesnt take so long - because its fast on local (what we thought we'd be getting, but are sus that it's not happening)
3. Increase the timeout - just basically masks the problem but still helps with outages

---

My plan today was to
1. create a new branch
2. revert our resource fix (e.g. add a bunch of resources to one sensor)
3. Deploy to production to see if the sensor falls over
4. If not introduce a change a new sensor that calls a no-op job does a more significant amount of processing (read all metadata files and log the contents of each)

**Comment by evantahler** [comment]
More deets in slack https://airbytehq-team.slack.com/archives/C059CTCUPDL/p1705615802664099

**Comment by erohmensing** [comment]
@erohmensing test

**Comment by erohmensing** [comment]
@erohmensing test

**Comment by erohmensing** [comment]
First test: [Adding back all resources as a dependency for one of the sensors](https://github.com/airbytehq/airbyte/commit/4d042891b2da3f70f5a7ba0d6de6d26dcad0bdd6). This fell over at one point, but was also probably on the cusp (as it fell over between version bumps and not due to a significant amount of new information we suspect we added. 

If we've gotten any new resources, we really hope this succeeds.

**Comment by erohmensing** [comment]
January 5, olddd metadata service code, sensor with all resources: 30s 
January 5 with a metadata service code update: pushes it over 1 min

![image](https://github.com/airbytehq/airbyte/assets/22333512/67a09ed2-3038-4124-aed4-4d346ced7cc1)

Remove extra resources: Down to ~7 seconds
![image](https://github.com/airbytehq/airbyte/assets/22333512/d974ee7d-7646-44ec-8af6-48a4a5a26376)

**Comment by erohmensing** [comment]
Added the original [heavy job](https://github.com/airbytehq/airbyte/commit/8680019052a7ac04e047c06786be88fddc1661ea) from [the PR where we introduced automaterialization](https://github.com/airbytehq/airbyte/pull/28190) - it still falls over. 
![image](https://github.com/airbytehq/airbyte/assets/22333512/692fd631-7183-49aa-ad90-ab8bf1e729b5)

In fact it looks like it falls over worse(?) than when we did the original implementation, since previously, the issue, from the loom at least, looks like it was that successful runs were making subsequent runs fail. Will try removing some of the github resourses to see if i can make that happen...

**Comment by erohmensing** [comment]
Tried to deploy the state before the Jan 5th deploy by checking it out, reverting the change, and then also cherrypicking the  pendulum change (otherwise it wouldn't deploy) but this didn't go anywhere, as airbyte-ci was pretty borked at the time.

## TL;DR
There was a LOT going on. We 
* didn't deploy for 3 months 
* deployed a lot of metadata-lib changes when deploying the new version
* had to work through a pendulum issue afterward
* had some sensor definitions that already had way too many (asset) resources it was requesting 

The sensor that once took 30seconds, then failed upon our deploy, now takes 30 seconds again. So that's... good, although its a bit unclear why it was working in the first place if it is really the new CPU that bumped it. In any case that sensor was reworked so that its down to 2 seconds (before and after the new CPU)

As a last ditch effort I'll check out the other sensors to see if they changed since Jan 16th.

**Comment by erohmensing** [comment]
Reverting the metadata-lib chagne without resetting the sensor keeps it short: 
![image](https://github.com/airbytehq/airbyte/assets/22333512/da531a2a-6345-4242-a67f-2148b2192354)

Resetting the sensor brings it back to 30s, not any faster
